### PR TITLE
feature/ProductImages

### PR DESCRIPTION
**Pull request by itsmeNEK**
Created UI
Created Image Carousel and lightbox

### CODE REVIEW

File: src/app/globals.scss [hunk 1]
```diff
@@ -6,7 +6,16 @@ body {
...
+.wrapper {
+  max-width: 1200px;
+  margin: 4rem auto 0 auto;
+  padding: 0 3rem;
 
+  @media screen and (max-width: 765px) {
+    padding: 0;
+    margin: 0;
+  }
+}
```
Comment (kbalagtey-work) [inline src/app/globals.scss:18]: do mobile-first approach

File: src/components/ProductImages/SubComponents/Carousel/Carousel.module.scss [hunk 2]
```diff
@@ -0,0 +1,57 @@
+.carousel_container {
+  display: flex;
+  position: relative;
+
+  &__content {
+    flex: 3;
+    text-align: center;
+  }
+  .prev_button,
+  .next_button {
```
Comment (kbalagtey-work) [inline src/components/ProductImages/SubComponents/Carousel/Carousel.module.scss:10]: if you are using BEM, then it is better to stick to the most common convention and use `-` to separate words.

block__element--modifier
my-block-name__element-name--modifier-name

Comment (itsmeNEK) [reply src/components/ProductImages/SubComponents/Carousel/Carousel.module.scss:10]: Thank you for the review. I will incorporate the suggested changes into my code.

File: src/components/ProductImages/SubComponents/Carousel/Carousel.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import Image from 'next/image'
+import Style from './Carousel.module.scss'
+import NavigationButton from '@/components/common/buttons/NavigationButton'
+import NextSvgIcon from '@/components/common/svg/NextSvgIcon'
+import PrevSvgIcon from '@/components/common/svg/PrevSvgIcon'
+import { useMediaQuery } from '@/hooks/useMediaQuery'
+import { TImage } from '@/types/Cart'
+
+type Props = {
```
Comment (kbalagtey-work) [inline src/components/ProductImages/SubComponents/Carousel/Carousel.tsx:9]: call it CarouselProps

as a convention

<NameOfComponent>Props

Comment (itsmeNEK) [reply src/components/ProductImages/SubComponents/Carousel/Carousel.tsx:9]: Alright, I will do that

File: src/types/Cart.ts [hunk 4]
```diff
@@ -1,3 +1,25 @@
...
+export type TImage = {
+  image: string
+  alt: string
+  small: boolean
+}
+export type TImages = {
+  smallImages: TImage[]
+  largeImages: TImage[]
+}
+export type TProduct = {
```
Comment (eremocal) [inline src/types/Cart.ts:10]: is there any reason why all of them are optional?

File: src/components/common/buttons/NavigationButton.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import React, { ButtonHTMLAttributes, ReactNode } from 'react'
```
Comment (eremocal) [inline src/components/common/buttons/NavigationButton.tsx:1]: Is there a difference between the `IconButton`?

**Pull request merged**